### PR TITLE
Implements inputs validator

### PR DESCRIPTION
**Pull request by rashidsp**
Implements Null and String validator for inputs.
inputs : ('feature_flag_key', 'experiment_key', 'user_id', 'variation_key', 'variable_key' and 'variable_type')
Methods:
-> Activate 
-> get_variation
-> Track
-> set_forced_variation through project_config-> set_forced_variation
-> get_forced_variation through project_config-> get_forced_variation
-> is_feature_enabled
-> get_enabled_features
-> get_feature_variable_for_type

### CODE REVIEW

**Comment by mikeng13** [comment]
Will be looking at this soon. Wanted to see if maybe we could run this through the compat suite

File: lib/optimizely/helpers/validator.rb [hunk 1]
```diff
@@ -95,6 +95,27 @@ def string_numeric?(str)
...
+        #
+        # logger - logger.
+        #
+        # Returns boolean True if all of the values are valid, False otherwise.
+
+        return false unless variables.respond_to?(:each) && !variables.empty?
+        is_valid = true
+        variables.each do |key, value|
+          next if value.is_a?(String) && !value.empty?
+          is_valid = false
```
Comment (thomaszurkan-optimizely) [inline lib/optimizely/helpers/validator.rb]: I don't think this is correct and your tests seem to confirm this.  If the last argument is invalid, everything is fine,  but, if your middle argument is invalid, it would not catch it.  Shouldn't it be is_valid ||= false?  Or return when you get the first is_valid = false.  Also, can you add tests that actually call the activate with a integer user id or a empty experiment key (first argument)?

Comment (thomaszurkan-optimizely) [inline lib/optimizely/helpers/validator.rb]: please put a break after setting is valid to false.
